File: Strings/__2000__/2535/2535.php
<?php

function inicial($inicial_raca, $nome) {
    foreach ($nome as $parte) {
        if ($parte[0] === $inicial_raca) {
            return true;
        }
    }
    return false;
}

while (($n = intval(readline())) !== 0) {

    $pets = 0;
    for ($i = 0; $i < $n; $i++) {
        $especie = trim(readline());
        $raca = trim(readline());
        $nome = explode(" ", trim(readline()));
        readline(); // linha em branco

        if (
            $especie === "cachorro" &&
            count($nome) > 1 &&
            inicial($raca[0], $nome)
        ) {
            $pets++;
        }
    }

    echo "$pets\n";
}

File: Strings/__2000__/2535/2535.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <vector>

bool isValidPet(const std::string& especie, const std::string& raca, const std::string& nome) {
    if (especie != "cachorro") return false;

    std::istringstream iss(nome);
    std::string palavra;
    std::vector<std::string> partes;

    while (iss >> palavra) {
        partes.push_back(palavra);
    }

    if (partes.size() < 2) return false; // precisa ser nome composto

    char inicialRaca = tolower(raca[0]);
    for (const auto& parte : partes) {
        if (tolower(parte[0]) == inicialRaca) {
            return true;
        }
    }

    return false;
}

int main() {
    int n;
    while (std::cin >> n) {
        std::cin.ignore(); // ignorar newline após o número
        int validos = 0;

        for (int i = 0; i < n; ++i) {
            std::string especie, raca, nome;

            std::getline(std::cin, especie);
            std::getline(std::cin, raca);
            std::getline(std::cin, nome);
            std::string linhaVazia;
            std::getline(std::cin, linhaVazia); // ler linha vazia

            if (isValidPet(especie, raca, nome)) {
                ++validos;
            }
        }

        std::cout << validos << std::endl;
    }

    return 0;
}